### PR TITLE
feat: add sources option to allow loading external component definition

### PR DESCRIPTION
**Pull request by stafyniaksacha**
This PR add a new `sources` option to allow loading pre-computed component meta file.

```ts
export default defineNuxtConfig({
  modules: [
    'nuxt-component-meta',
  ],
  componentMeta: {
    sources: [
      // import from a module
      '@nuxt/ui/component-meta',
      // resolve in project aliases
      '~/meta/library',
      // or directly raw input
      {
        NInput: {
          //...
        },
      }
    ],
  },
})
```

(This does not cover sources generation, but I have a naive approach [here](https://github.com/shuriken-ui/nuxt/blob/main/package.json#L24-L28) with [this](https://github.com/shuriken-ui/nuxt/blob/main/scripts/copy-meta.ts))

### CODE REVIEW

**Comment by stafyniaksacha** [comment]
First, I need to talk about the issue.

When using a module living in `node_modules`, vue-component-meta can't resolve types of exposed components due to `.nuxt/tsconfig.json` exclude.

This is a basic example:
```vue
<script setup lang="ts">
const meta = await useComponentMeta('UButton')
</script>

<template>
  <div>
    <UButton>Button</UButton>
    <!-- props, slots, events, exposed are all empty -->
    <pre>{{ meta.meta }}</pre>
  </div>
</template>
```
> You can find a working version here: https://github.com/stafyniaksacha/nuxt-component-meta-deep-repro

Also, to me, it does not make sense to scan those components each time we start our nuxt app, as they are not likely to change during development.

The solution I've imagined is that when publishing a component library, we can include those pre-computed meta into the package (at the cost of package size, but we can publish to another meta package).

This PR aims to be able to load and merge pre-computed meta into our app, so we can make our `useComponentMeta('UButton')` working as expected without needing to parse them.

In order to generate the pre-computed meta in our component library, we can't simply add the `nuxt-meta-component` because it will affect end-users, instead I made a poc for a CLI using `citty`. 
The idea behind is to dynamically register `nuxt-component-meta` module and do a build. This will generate `.nuxt/component-meta.mjs` and `.nuxt/component-meta.d.ts`, we can then either publish them within the package or in another one.

> You will find the CLI here: https://github.com/stafyniaksacha/nuxt-meta-component-cli/blob/main/src/main.ts#L33-L75

I think this CLI should be added into this repo if you are ok with the idea.

Then, we can update our component library modules to register those meta if `nuxt-component-meta` module is detected.

**Comment by farnabaz** [comment]
Having pre-compiled meta is a good idea to fix typescript exclude issue, and also it will improve performance of development and build.

Having a CLI to generate meta is also a good Idea. Personally, I would like to ship these pre-compiled meta within the components package and not as a different package, therefore I think we might define an output directory for CLI to write generated meta files. This also make it easier for developers to use the CLI and publish meta as separate package.

Additionally, we can create a hook (`component-meta:extend`) which modules could use it to add/extend components metadata.

Looking forward to seeing the CLI implementation on your PR :)

File: src/module.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export default defineNuxtModule<ModuleOptions>({
     rootDir: nuxt.options.rootDir,
     componentDirs: [],
     components: [],
+    sources: [],
```
Comment (farnabaz) [inline src/module.ts:29]: I think using `metaSources` might be a better name for this option and will be consistent with `metaFields` 

Also, CLI should generate all `metaFields` and module can filter them before using them.

**Comment by pi0** [comment]
In addtion, we might find a hook in nuxt module builder to precompute this meta after mkdist job to prebundle them 👍🏼

**Comment by stafyniaksacha** [comment]
@farnabaz @pi0 The cli is ready for review!

You can use it with:

```bash
# build with unbuild
pnpm prepack

# generate meta from playground
./bin/nuxt-component-meta.mjs playground --no-schema
```

There is an issue with the prop type detection into the playground, comment module registration in the nuxt config like: 
```ts
  modules: [
    '@nuxt/content',
    'pinceau/nuxt',
    // nuxtMetaModule as any
  ],
```

I think it's due to typescript that can not resolve typing, it's working great when targeting a project that already have older version of this module

File: bin/nuxt-component-meta.mjs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env node
+
+import { runGenerate } from '../dist/runtime/cli/index.mjs'
```
Comment (pi0) [inline bin/nuxt-component-meta.mjs]: CLI entrypoint should not be in `dist/runtime` (it is a build time entry)

Comment (pi0) [inline bin/nuxt-component-meta.mjs]: CLI entrypoint should not be in `dist/runtime` (it is a build time entry)

Comment (stafyniaksacha) [reply bin/nuxt-component-meta.mjs]: @pi0, I think you reviewed an old commit, I moved it to `'../dist/cli.mjs'` using a new build.entries in package.json

Comment (stafyniaksacha) [reply bin/nuxt-component-meta.mjs]: Do you think it's relevant to have a nuxt cli kit that would export those kinds of utils: https://github.com/nuxtlabs/nuxt-component-meta/pull/60/files#diff-280cd3ea2fd76f9dc66e2878205eed73fd8006764453fee4c1f08742d4840f2d

Comment (pi0) [reply bin/nuxt-component-meta.mjs]: Yes i was thinking CLI could expose kit/nuxt loading utils also had other usecases 👍🏼

**Comment by stafyniaksacha** [comment]
@farnabaz any chance to see it merged?

**Review by farnabaz — APPROVED** [review]
Sorry for late response.
LGTM ❤️ 🚀

**Pull request merged**